src/colmap/geometry/gps.h

```diff
@@ -44,7 +44,8 @@ namespace colmap {
 // representation and vice versa.
 class GPSTransform {
  public:
-  MAKE_ENUM(ELLPSOID, 0, GRS80, WGS84);
+  MAKE_ENUM(Ellpsoid, 0, GRS80, WGS84);
+  MAKE_ENUM(CartesianFrame, 0, ECEF, ENU, UTM);
```

Comment:
Would it make sense to merge both of these enums with `PosePrior::CoordinateSystem`? The pose priors could be equivalently in ECEF or GRS80 (but not UTM since it'd need to include the zone).

Reply:
The decomposition into Ellipsoid and CartesiamFrame seems to be a better one than the one in `PosePrior::CoordinateSystem`. One can use ECEF/ENU/UTM cartesian coordinates based on different ellipsoids. Is your proposal to replace the `PosePrior::CoordinateSystem` with the two enums here or flatten it into an enum with 6 values?

Reply:
Looking at this again, I think the following serves all the practical use cases:
```
  MAKE_ENUM_CLASS(CoordinateSystem,
                  -1,
                  UNDEFINED,        // = -1
                  SPHERICAL_WGS84,  // = 0
                  CARTESIAN,        // = 1
                  SPHERICAL_GRS80,  // = 2
  );
```
With this, one can store lat/lon/alt either wrt to GRS80/WGS84 or directly in a suitable cartesian frame but we don't really care about which frame that is as long as it is cartesian. If we later care about whether the cartesian frame is metric, we could distinguish between `CARTESIAN_METRIC` and `CARTESIAN_NON_METRIC` later.

Reply:
@ahojnnes In my opinion, `CARTESIAN` is a high-level abstraction, while `SPHERICAL_WGS84` and `SPHERICAL_GRS80` are specific and accurate spherical frames. Should they be in the same enum class?

If we want to make one enum only, maybe this is better?
```cpp
MAKE_ENUM_CLASS(-1, 
                UNDEFINED, 
                SPHERICAL_WGS84, 
                SPHERICAL_GRS80,
                CARTESIAN_ECEF, 
                CARTESIAN_ENU,
                CARTESIAN_UTM);
```

Or more abstract, 
```cpp
MAKE_ENUM_CLASS(-1, 
                UNDEFINED, 
                SPHERICAL, 
                CARTESIAN);
```

Reply:
Currently, we cannot fully represent/serialize all information in CARTSIAN_UTM/CARTESIAN_ENU in the pose prior, because we lose the zone and origin. I am also unsure what we would do with this information.

Let's also be careful to preserve the existing enum values, otherwise existing database content will be broken.

Reply:
Since we are expanding the cartesian frames to include ECEF and UTM, I think that we should make the reference frame of each pose prior explicit (instead of only whether it is cartesian or not). This would allow:
- having databases with both WGS84 and cartesian priors (such that we can convert from the former to the latter)
- making a difference between pose priors in cartesian frames that are gravity-aligned (ENU, UTM) and those that are not (ECEF), which will be required when integrating IMU or image-based gravity priors.

Thus I indeed suggest to have an enum with 6 values, e.g. UNDEFINED, WGS84, GRS80, ECEF, ENU, UTM. The only problem is that UTM is not sufficient on its own since it needs a zone...

Alternatively, if we want to enforce that all pose priors are provided in the same reference frame, then we should make this an attribute of the database table rather than of each row.

Reply:
I understand, so lat/lon/alt and ECEF(or say CARTESIAN) are worth being abstracted out separately, because they both represent sufficient information with just a three-dimensional vector. :D

`GPSTransform` is a helper class to convert gps coordinates, maybe the enum inside it could be more specific.

Reply:
In other words, I propose to go with (keeping it backwards compatible with old enum values):
```
MAKE_ENUM_CLASS(CoordinateSystem, -1, UNDEFINED, SPHERICAL_WGS84, CARTESIAN_ECEF , SPHERICAL_GRS80, CARTESIAN_ENU, CARTESIAN_UTM);
```